smart-contracts/contracts/merkle-incentives/src/incentives/helpers.rs

```diff
@@ -4,329 +4,223 @@ use crate::{
-                    135, 55, 235, 99, 212, 99, 110, 234, 74, 171, 238, 24, 97, 234, 24, 208,
-                ]),
-            },
-            Entry {
-                is_left_sibling: false,
-                hash: Hash::from(vec![
-                    238, 188, 213, 212, 248, 3, 198, 33, 124, 55, 104, 81, 71, 157, 143, 217, 27,
-                    30, 233, 138, 120, 2, 27, 118, 47, 174, 142, 226, 67, 100, 218, 195,
-                ]),
-            },
-            Entry {
-                is_left_sibling: false,
-                hash: Hash::from(vec![
-                    51, 5, 59, 42, 196, 9, 23, 37, 150, 141, 131, 3, 195, 252, 188, 219, 218, 114,
-                    15, 107, 219, 144, 56, 214, 22, 12, 158, 92, 20, 215, 255, 214,
-                ]),
-            },
-        ];
-        println!("proof_bytes{:?}", proof_bytes);
+    // Test constants
+    const MERKLE_ROOT_STRING: &str = "0hGvbH+l9pdPgOmJY6wZuwjsrvtPsuslgTURavrUP6I=";
+    const MERKLE_ROOT_INVALID_STRING: &str = "INVALIDROOTRC3R6LzoFpYmMJ81IUY5nTVr+X5/OsXI=";
 
-        let proof_string_hash = serde_json_wasm::to_string(&proof_bytes).unwrap();
-        println!("proof_string{:?}", proof_string_hash);
+    const USER_ADDRESS: &str = "osmo1cn2t4zha4ukq42u2q8x0zgyp60hp5gy54a2wxt";
 
-        let _: Proof = serde_json_wasm::from_str(proof_string_hash.as_str()).unwrap();
-        let merkle_root: &str = "yZsdqUjU3p9Iy7Df1Eon5Op+2d08cHVokMX5bvA3I00=";
+    const CLAIM_PROOF_STRING: &str = "osmo1cn2t4zha4ukq42u2q8x0zgyp60hp5gy54a2wxt900000000ugauge";
+    const CLAIM_PROOF_INVALID_STRING: &str =
+        "osmo1cn2t4zha4ukq42u2q8x0zgyp60hp5gy54a2wxt999999999ugauge";
 
-        verify_proof(
-            &merkle_root.to_string(),
-            proof_string_hash.as_str(),
-            "osmo1cn2t4zha4ukq42u2q8x0zgyp60hp5gy54a2wxt900000000uosmo",
-        )
-        .unwrap();
-        //println!("proof_struct{:?}", proof_struct);
-    }
+    const CLAIM_AMOUNT: u128 = 900000000;
+    const CLAIM_AMOUNT_INVALID: u128 = 999999999;
 
-    #[test]
-    fn test_verify_success() {
-        // this test is taken directly from the testdata. See the README.md of this contract
-        verify_proof(
-            &MERKLE_ROOT_STRING.to_string(),
-            USER_MERKLE_PROOF,
-            TO_VERIFY,
-        )
-        .unwrap();
-    }
-
-    #[test]
-    fn test_verify_err() {
-        // this test is taken directly from the testdata. See the README.md of this contract
-        let invalid_merkle_root = "INVALIDROOTRC3R6LzoFpYmMJ81IUY5nTVr+X5/OsXI=";
-
-        let result = verify_proof(
-            &invalid_merkle_root.to_string(),
-            USER_MERKLE_PROOF,
-            TO_VERIFY,
-        );
-
-        if let Err(ContractError::FailedVerifyProof {}) = result {
-            assert!(true); // expected
-        } else {
-            panic!("unexpected result");
+    // Utils functions
+    fn get_proof_hashes() -> Vec<[u8; 32]> {
```

Comment:
This test would be cleaner if instead we would reuse the already imported merkle crate, create a tree using that crate and call our code for a leaf in that trait. If this current tree comes is generated by the backend server, we can leave it for now so that we still have some kind of logic connection between our contract and the backend server and add a better E2E test later.

Reply:
Yes this tree is generated by our backend infra, but still you are right as this is good to have, but not excluding internal testing. There is a lack of testing using the smart contract used crate, so I agree that creating a full Rust test case is needed.

Reply:
@njerschow covered this with a new test case that has been merged from #575